src/typescript/06-types/types.md

```diff
@@ -264,15 +264,15 @@ This solves our function signature problem and allows us to pass any collection
 
 ### Object Creation
 
-The first problem comes when creating a dinosaur object. Union types simplify creating an object – all you need to do is pass it keys from any of the types in the union. The only keys that are required are the ones that are shared across all types in the union. In our case, the only shared key across `LandDinosaur`, `AirDinosaur`, and `WaterDinosaur` is `name`; anything else is fair game to add as long as the keys are from the same type. However, this means we can get type-safe objects that don't make a lot of sense. Take, for example, this pterodactyl object.
+The first problem comes when creating a dinosaur object. Union types simplify creating an object – we need to pass it keys from any of the types in the union. The only keys that are required are the ones that are shared across all types in the union. In our case, the only shared key across `LandDinosaur`, `AirDinosaur`, and `WaterDinosaur` is `name`; anything else is fair game to add as long as the keys are from the same type. However, this means we can get type-safe objects that don't make a lot of sense. Take, for example, this pterodactyl object.
```

Comment:
This en dash is almost right! It should be an em dash. `–` -> `—`.

I would break up any of these longer paragraphs into more digestible bits. This one could be split at “In our case,” and probably at “However” too.

This paragraph should also end with a colon because it’s pointing to the code sample below.